src/common/meta/src/ddl/create_flow.rs

```diff
@@ -0,0 +1,259 @@
+        Ok(Status::executing(true))
+    }
+
+    /// Creates flow task metadata.
+    ///
+    /// Abort(not-retry):
+    /// - Failed to create table metadata.
+    async fn on_create_metadata(&mut self) -> Result<Status> {
+        // Safety: The flow task id must be allocated.
+        let flow_task_id = self.data.flow_task_id.unwrap();
+        // TODO(weny): Support `or_replace`.
+        self.context
+            .flow_task_metadata_manager
+            .create_flow_task_metadata(flow_task_id, self.data.to_flow_task_info_value())
+            .await?;
+        info!("Created flow task metadata for flow task {flow_task_id}");
+        Ok(Status::done_with_output(flow_task_id))
+    }
+}
+
+#[async_trait]
+impl Procedure for CreateFlowProcedure {
+    fn type_name(&self) -> &str {
+        Self::TYPE_NAME
+    }
+
+    async fn execute(&mut self, _ctx: &ProcedureContext) -> ProcedureResult<Status> {
+        let state = &self.data.state;
+
+        let _timer = metrics::METRIC_META_PROCEDURE_CREATE_FLOW_TASK
+            .with_label_values(&[state.as_ref()])
+            .start_timer();
+
+        match state {
+            CreateFlowTaskState::Prepare => self.on_prepare().await,
+            CreateFlowTaskState::FlownodeCreateFlows => self.on_flownode_create_flow().await,
+            CreateFlowTaskState::CreateMetadata => self.on_create_metadata().await,
+        }
+        .map_err(handle_retry_error)
+    }
+
+    fn dump(&self) -> ProcedureResult<String> {
+        serde_json::to_string(&self.data).context(ToJsonSnafu)
+    }
+
+    fn lock_key(&self) -> LockKey {
+        let catalog_name = &self.data.task.catalog_name;
+        let task_name = &self.data.task.task_name;
+        let schema_names = self
+            .data
+            .task
+            .source_table_names
+            .iter()
+            .map(|name| &name.schema_name)
+            .chain(std::iter::once(&self.data.task.sink_table_name.schema_name))
+            .collect::<HashSet<_>>();
+        let source_table_names = &self.data.task.source_table_names;
+        let mut keys = Vec::with_capacity(2 + schema_names.len() + source_table_names.len());
+        keys.extend([
+            CatalogLock::Read(catalog_name).into(),
+            FlowTaskNameLock::new(catalog_name, task_name).into(),
+        ]);
+        for name in schema_names {
+            keys.push(SchemaLock::read(catalog_name, name).into());
+        }
+        for name in source_table_names {
+            keys.push(
+                TableNameLock::read(&name.catalog_name, &name.schema_name, &name.table_name).into(),
+            )
+        }
+        LockKey::new(keys)
+    }
+}
+
+/// The state of [CreateFlowTaskProcedure].
+#[derive(Debug, Clone, Serialize, Deserialize, AsRefStr, PartialEq)]
+pub enum CreateFlowTaskState {
+    /// Prepares to create the flow.
+    Prepare,
+    /// Creates flows on the flownode.
+    FlownodeCreateFlows,
```

Comment:
```suggestion
    CreateFlows,
```

I think `CreateFlow` is OK